### DISCUSSION TITLE
How to delete ISR cache for deleted pages?

### DISCUSSION THREAD

**Discussion by makutamoto** [vercel/next.js#21055] [Help]
I'm using ISR to create a site let users post contents.
The following is my code.
```typescript
export const getStaticProps: GetStaticProps<Props> = async ({ params }) => {
    const res = await invokeQueryWithApiKey(GetProblem, {
        authorUsername: params.username || '',
        problemSlug: params.problemSlug || '',
    })
    if (res.user === null || res.user.problem === null) {
        return {
            notFound: true,
        }
    }
    return {
        props: {
            user: res.user,
        },
        revalidate: 1,
    }
}

export const getStaticPaths: GetStaticPaths = async () => ({
    paths: [],
    fallback: 'blocking',
})
```
When the user visits pages not exist, my site responds with 404 not found as I expected.
However when the user visits deleted pages, my site returns their cache.
So I want to delete them.
Is there any way?

**Comment by vassbence** [comment]
Yes, return `notFound`. This is a recent addition to fix similar issues you have just mentioned, it basically purges the cache and returns a 404 even if there once was a successful `getStaticProps` run.

**Reply by makutamoto** [reply]
I tried it again.
![image](https://user-images.githubusercontent.com/34860673/104831867-e649c180-58cf-11eb-9385-58fcfa1d187c.png)
But my site shows a cache, although my log shows 404 status.
Does it take a long time until the new feature you mentioned has been deployed?
